src/python/pants/option/global_options.py

```diff
@@ -1747,17 +1748,31 @@ def validate_instance(cls, opts):
                 )
             )
 
-        if opts.remote_execution and (opts.remote_cache_read or opts.remote_cache_write):
-            raise OptionsError(
-                softwrap(
-                    """
-                    `--remote-execution` cannot be set at the same time as either
-                    `--remote-cache-read` or `--remote-cache-write`.
+        deprecated_conditional(
+            lambda: opts.remote_execution and opts.remote_cache_eager_fetch,
+            removal_version="2.15.0.dev0",
+            entity="Setting `--remote-execution` at the same time as `--remote-cache-eager-fetch`.",
+            hint=softwrap(
+                """
+                Disable `--remote-cache-eager-fetch`.
+
+                Eagerly fetching from the remote cache only makes sense with a local executor.
+                """
+            ),
+        )
 
-                    If remote execution is enabled, it will already use remote caching.
-                    """
-                )
-            )
+        deprecated_conditional(
+            lambda: opts.remote_execution
+            and (not opts.remote_cache_read or not opts.remote_cache_write),
+            removal_version="2.15.0.dev0",
+            entity="Using `--remote-execution` without `--remote-cache-read` and `--remote-cache-write`.",
+            hint=softwrap(
+                """
+                Use of `--remote-execution` implies use of a remote cache. Both
+                `--remote-cache-read` and `--remote-cache-write` should be set as well.
+                """
+            ),
```

Comment:
Please fix our remote execution docs in this PR, or immediately thereafter in a follow-up.

Reply:
Will do. It seems like there is still some pending discussion of these two deprecations though, so I'll hold off until we've agreed on that. cc @tdyas